pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1429,9 +1460,41 @@ COMMENT ON FUNCTION SCHEMA_CATALOG.epoch_abort(BIGINT)
 IS 'ABORT an INSERT transaction due to the ID epoch being out of date';
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.epoch_abort TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_used_series(
```

Comment:
since the set if potential_series ids is much smaller than the set of unique series I suggest something like the following instead 
```
CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_confirmed_unused_series(
    series_table TEXT, potential_series_ids BIGINT[]
) RETURNS TABLE (confirmed_series_id BIGINT[]) 
AS $func$
DECLARE
    r RECORD;
BEGIN
    FOR r IN
        SELECT *
        FROM SCHEMA_CATALOG.metric m
        WHERE m.series_table = get_used_series.series_table
    LOOP

    --at each iteration of the loop filter potential_series_ids to only
    --have those series ids that don't exist in the metric tables.
    EXECUTE format(
    $query$
        SELECT array_agg(potential_series.series_id)
				FROM unnest($1) as potential_series(series_id)
				LEFT JOIN LATERAL(
					SELECT 1
				  FROM  %1$I.%2$I
				  WHERE data_exists.series_id = potential_series.series_id
				  --use chunk append + more likely to find something starting at earliest time
				  ORDER BY time ASC
				  LIMIT 1
				) as lateral_exists(indicator) ON (true)
				WHERE lateral_exists.indicator IS NULL
    $query$, r.table_schema, r.table_name)
		USING potential_series_ids
    INTO potential_series_ids;

    END LOOP;
    RETURN potential_series_ids;
END
$func$
LANGUAGE PLPGSQL VOLATILE
```

Which could be used like 
```
SELECT SCHEMA_CATALOG.get_confirmed_unused_series('%1$s', array_agg(series_id))
FROM potentially_drop_series
```

By the way I think the table name parameter should be the series_table not the metric_table. Conceptually, the loop should be all the metric tables that use the same series_table. I also don't think we need the check time in this function.